### PR TITLE
Save button only enabled if form is filled out correctly

### PR DESCRIPTION
**Pull request by melanieretter**
The save button in create/update dialogs is only enabled if all mandatory fields are filled out, are valid and if changes exists (in update case)

savebuttondisabled

### CODE REVIEW

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/layouts/AbstractCreateUpdateTagLayout.java [hunk 1]
```diff
@@ -644,4 +596,20 @@ public FormLayout getFormLayout() {
         return formLayout;
     }
 
+    public GridLayout getMainLayout() {
+        return mainLayout;
+    }
+
+    @Override
+    public void addColorChangeListener(final ColorChangeListener listener) {
+        // TODO Auto-generated method stub
```
Comment (schabdo) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/layouts/AbstractCreateUpdateTagLayout.java]: remove TODO

Comment (Jonkno) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/layouts/AbstractCreateUpdateTagLayout.java]: remove TODO

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/layouts/AbstractCreateUpdateTagLayout.java [hunk 2]
```diff
@@ -644,4 +596,20 @@ public FormLayout getFormLayout() {
...
+
+    }
+
+    @Override
+    public void removeColorChangeListener(final ColorChangeListener listener) {
+        // TODO Auto-generated method stub
```
Comment (Jonkno) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/layouts/AbstractCreateUpdateTagLayout.java]: remove TODO

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/AddUpdateRolloutWindowLayout.java [hunk 3]
```diff
@@ -691,15 +668,18 @@ public void validate(final Object value) {
...
+                    new IntegerRangeValidator(i18n.get(MESSAGE_ROLLOUT_FIELD_VALUE_RANGE, 0, 500), 0, 500)
+                            .validate(Integer.valueOf(value.toString()));
+                }
             }
             // suppress the need of preserve original exception, will blow
             // up the
             // log and not necessary here
             catch (@SuppressWarnings("squid:S1166") final InvalidValueException ex) {
-                LOG.error(ex.getMessage());
+                // we have to throw the exception here, otherwise the UI won't
```
Comment (schabdo) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/AddUpdateRolloutWindowLayout.java]: You do not suppress the exception anymore. So you can remove the `@SuppressWarnings`

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/AddUpdateRolloutWindowLayout.java [hunk 4]
```diff
@@ -672,15 +646,18 @@ private int getGroupSize() {
...
+                    new IntegerRangeValidator(i18n.get(MESSAGE_ROLLOUT_FIELD_VALUE_RANGE, 0, 100), 0, 100)
+                            .validate(Integer.valueOf(value.toString()));
+                }
             }
             // suppress the need of preserve original exception, will blow
             // up the
             // log and not necessary here
             catch (@SuppressWarnings("squid:S1166") final InvalidValueException ex) {
-                LOG.error(ex.getMessage());
+                // we have to throw the exception here, otherwise the UI won't
```
Comment (schabdo) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/AddUpdateRolloutWindowLayout.java]: remove sonar rule suppression

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/AddUpdateRolloutWindowLayout.java [hunk 5]
```diff
@@ -622,42 +600,38 @@ private TextField createRolloutNameField() {
...
                 }
             }
             // suppress the need of preserve original exception, will blow
             // up the
             // log and not necessary here
-            catch (@SuppressWarnings("squid:S1166") final InvalidValueException ex) {
-                LOG.error(ex.getMessage());
+            catch (
+
+            @SuppressWarnings("squid:S1166") final InvalidValueException ex) {
```
Comment (schabdo) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/AddUpdateRolloutWindowLayout.java]: remove sonar rule suppression

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/AddUpdateRolloutWindowLayout.java [hunk 6]
```diff
@@ -622,42 +600,38 @@ private TextField createRolloutNameField() {
...
-
     class ErrorThresoldOptionValidator implements Validator {
         private static final long serialVersionUID = 9049939751976326550L;
 
         @Override
         public void validate(final Object value) {
             try {
                 if (HawkbitCommonUtil.trimAndNullIfEmpty(noOfGroups.getValue()) == null
-                        || HawkbitCommonUtil.trimAndNullIfEmpty((String) targetFilterQueryCombo.getValue()) == null) {
+                        || (HawkbitCommonUtil.trimAndNullIfEmpty((String) targetFilterQueryCombo.getValue()) == null
```
Comment (schabdo) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/AddUpdateRolloutWindowLayout.java]: The condition is hard to understand. Please extract a method and provide a proper name

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/CommonDialogWindow.java [hunk 7]
```diff
@@ -104,6 +203,154 @@ private final void init() {
         center();
         setModal(true);
         addStyleName("fontsize");
+        setOrginaleValues();
+        addListeners();
+    }
+
+    public final void setOrginaleValues() {
```
Comment (Jonkno) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/CommonDialogWindow.java]: Add javadoc

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/AddUpdateRolloutWindowLayout.java [hunk 8]
```diff
@@ -147,22 +143,24 @@
...
         buildLayout();
     }
 
-    public CommonDialogWindow getWindow() {
-
-        addUpdateRolloutWindow = SPUIComponentProvider.getWindow(i18n.get("caption.configure.rollout"), null,
-                SPUIDefinitions.CREATE_UPDATE_WINDOW, this, event -> onRolloutSave(), event -> onDiscard(),
-                uiProperties.getLinks().getDocumentation().getRolloutView());
-        return addUpdateRolloutWindow;
+    public CommonDialogWindow getWindow(final Long rolloutId) {
```
Comment (schabdo) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/AddUpdateRolloutWindowLayout.java]: I saw a couple of calls like `getWindow(null)`. This is a good indicator that this method is doing to much.
Is it feasable to provide a second one like:

`getWindow()` which do not call `populateData(...)` or call `getWindow(null)` internal.

**Pull request merged**